File: Data_Preparation/Preprocessing.py
## Taking the csv data save it as hdf5 and preprocess it for UMAP and HDBSCAN ##
## Import csv file with normalized counts and save as hdf5 file
import pandas as pd
data = pd.read_csv('/netscratch/dep_mercier/grp_novikova/laura/data/Single_Cell/Root_Atlas_normalized_counts.csv',index_col=0)
##### Get the corresponding celltypes
data = data.transpose()
data.to_hdf('/netscratch/dep_mercier/grp_novikova/laura/data/Single_Cell/RNA-Seq.h5', key='counts')

File: Data_Preparation/Preprocessing.R
# Save Root Atlas normaliced count data as csv-file
# Necessary Libraries
library(Seurat)
# Data Analysis
rc.integrated <- readRDS('/netscratch/dep_mercier/grp_novikova/laura/data/Single_Cell/GSE152766_Root_Atlas.rds') #Load data 
normalized_counts <- as.matrix(rc.integrated@assays$integrated@data) #extract normalized counts
celltypes_shahan <- as.matrix(rc.integrated@meta.data$celltype.anno) #extract annotation of Shahan et al.

write.csv(normalized_counts, "/netscratch/dep_mercier/grp_novikova/laura/data/Single_Cell/Root_Atlas_normalized_counts.csv", row.names=TRUE, col.names=TRUE) #Save counts as csv to import it with python 
write.csv(celltypes_shahan, "~/data/Single_Cell/Celltypes_Shahan.csv")